source/views/Options/Views/Welcome/steps/ImportMnemonicStep/index.jsx

```diff
@@ -0,0 +1,161 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import * as bip39 from 'bip39';
+
+import { HANDLER_TYPES, sendMessage } from '@background/Keyring';
+import { clearStorage } from '@modules/storageManager';
+import { getRandomEmoji } from '@shared/constants/emojis';
+import { createArray } from '@shared/utils/array';
+import { Alert, Button, FormInput } from '@components';
+import facepalmEmoji from '@assets/icons/facepalm.svg';
+
+import useStyles from './styles';
+import MnemonicInput from '../../components/MnemonicInput';
+
+const MNEMONIC_LENGTH = 12;
+
+export function clearClipboard() {
+  window.navigator.clipboard.writeText('');
+}
+
+const isValidMnemonic = (mnemonic) => bip39.validateMnemonic(mnemonic);
+
+const ImportWalletStep = ({ handleNextStep, handleSetMnemonic }) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const [draftMnemonic, setDraftMnemonic] = useState(
+    createArray(MNEMONIC_LENGTH, ''),
+  );
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+  const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+
+  const [invalidMnemonic, setInvalidMnemonic] = useState(true);
+
+  const onMnemonicChange = (newDraftMnemonic) => {
+    let newMnemonicError = '';
+    const joinedDraftMnemonic = newDraftMnemonic.join(' ').trim().toLowerCase();
+    if (newDraftMnemonic.some((word) => word !== '')) {
+      if (newDraftMnemonic.some((word) => word === '')) {
+        newMnemonicError = t('seedPhraseReq');
+      } else if (!isValidMnemonic(joinedDraftMnemonic)) {
+        setInvalidMnemonic(true);
+        newMnemonicError = t('invalidSeedPhrase');
+      }
+    }
+    setInvalidMnemonic(newMnemonicError?.length);
+    setDraftMnemonic(newDraftMnemonic);
+    handleSetMnemonic(newMnemonicError ? '' : newDraftMnemonic);
+  };
+
+  const handleCreateAccount = async () => {
+    // clean the storage before initiating keyring
+    clearStorage();
+    const params = {
+      password,
+      mnemonic: draftMnemonic.join(' '),
+      icon: getRandomEmoji(),
+    };
+    sendMessage({ type: HANDLER_TYPES.IMPORT, params }, (response) => {
+      handleSetMnemonic(response?.mnemonic);
+    });
+    handleNextStep();
+  };
+
+  const onPasswordChange = useCallback(
+    (event) => {
+      const newPassword = event.target.value;
+      let newConfirmPasswordError = '';
+      let newPasswordError = '';
+
+      if (newPassword && newPassword.length < 8) {
```

Comment:
we should move that "8" to a constant in a general folder, and reuse it here and in `CreatePasswordStep`